File: src/gof/Builder/code/MazeFactory.cpp
class MazeFactory {
public:
    MazeFactory();
    
    virtual Maze* MakeMaze() const
        { return new Maze; }
    virtual Wall* MakeWa11() const
        { return new Wall; }
    virtual Room* MakeRoom( int n) const
        { return new Room(n) }
    virtual Door* MakeDoor (Room* rl, Roam* r2) const
        { return new Door (rl, r2) };
}

File: src/gof/Builder/code/MazeFactory.java
public class MazeFactory {
    
    public Maze makeMaze(){
         return new Maze(); 
    }
    public Wall makeWa11(){
         return new Wall(); 
    }
    public Room makeRoom(int n){
         return new Room(n);
    }
    public Door makeDoor (Room rl, Roam r2){
         return new Door(rl, r2);
    }
}